enricher/api/src/main/java/io/fabric8/maven/enricher/api/BaseEnricher.java

```diff
@@ -133,8 +134,30 @@ protected void addInitContainer(PodTemplateSpecBuilder builder, JSONObject initC
         String initContainerAnnotation = builder.buildMetadata().getAnnotations().get(INIT_CONTAINER_ANNOTATION);
         JSONArray initContainers = Strings.isNullOrBlank(initContainerAnnotation) ? new JSONArray()
                 : new JSONArray(initContainerAnnotation);
-        initContainers.put(initContainer);
 
+        // lets avoid duplicates being added
+        boolean duplicate = false;
+        int length = initContainers.length();
+        for (int i = 0; i < length; i++) {
+            JSONObject obj = initContainers.getJSONObject(i);
+            if (duplicateInitcontainer(obj, initContainer)) {
+                duplicate = true;
+            }
+
+        }
+        if (!duplicate) {
+            initContainers.put(initContainer);
+        }
         builder.editMetadata().addToAnnotations(INIT_CONTAINER_ANNOTATION, initContainers.toString()).endMetadata();
     }
+
+    private boolean duplicateInitcontainer(JSONObject a, JSONObject b) {
+        Object name1 = a.get("name");
+        Object name2 = b.get("name");
+        if (Objects.equal(name1, name2)) {
+            log.warn("Found duplicate init containers with names " + name1 + " so ignoring " + b);
```

Comment:
No enricher should fired again on already enriched resources introduced by the `DependenciesEnricher`. That must be fixed, that's the root cause.

And if this is fixed, I don't get why the first init container should be used and not the second one (if they differ) ? And why not stop with an error because at least one init container won't run ? Why not renaming the second init container to something unique ?

The fix of course will prevent the build from breaking, but does it not also allow for inconsistencies and bugs to creep in silently (or buried in mega bytes of logs) ?

I really would love to have a more rigour discussion of this kind before merging (guess, that's what reviews are for). 

I'm not against quick fixing a pressing issue with a patch, but maybe we can agree to open a new issue for looking to fix the real cause ?